Assets/LeapMotionModules/UI/Interaction/InteractionBehaviourBase.cs

```diff
@@ -0,0 +1,130 @@
+﻿using Leap;
+using Leap.Unity.Attributes;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Leap.Unity.UI.Interaction {
+
+  public abstract class InteractionBehaviourBase : MonoBehaviour {
+
+    public InteractionManager interactionManager;
+
+    protected Rigidbody _body;
+    public Rigidbody Rigidbody { get { return _body; } }
+
+    [SerializeField]
+    [HideInInspector]
+    #pragma warning disable 0414
+    // Variable used only for inspector rendering.
+    private bool _interactionManagerIsNull = true;
+    #pragma warning restore 0414
+
+    [Header("Interaction Settings")]
+    [DisableIf("_interactionManagerIsNull", isEqualTo: true)]
+    public bool ignoreHover = false; // TODO: 
+    [DisableIf("_interactionManagerIsNull", isEqualTo: true)]
+    public bool ignoreContact = false; // TODO: Contact NYI.
+    [DisableIf("_interactionManagerIsNull", isEqualTo: true)]
+    public bool ignoreGrasping = false;
+    /// <summary> Can this object be grasped with two or more hands? </summary>
+    [DisableIf("_interactionManagerIsNull", isEqualTo: true)]
+    public bool allowMultiGrasp = false;
+
+    /// <summary>
+    /// Called by the InteractionManager every FixedUpdate, after
+    /// InteractionHands have FixedUpdated.
+    /// 
+    /// InteractionBehaviour uses this to fire per-object interaction events.
+    /// </summary>
+    public abstract void FixedUpdateObject();
+
+    protected virtual void Awake() {
+      _body = GetComponent<Rigidbody>();
+      _body.maxAngularVelocity = 100F;
+
+      if (interactionManager != null) {
+        interactionManager.RegisterInteractionBehaviour(this);
+      }
+    }
+
+    protected virtual void Start() {
+      if (interactionManager == null) {
+        interactionManager = InteractionManager.singleton;
+
+        if (interactionManager == null) {
+          Debug.LogError("Interaction Behaviours require an Interaction Manager. Please ensure you have an InteractionManager in your scene.");
+        }
+        else {
+          interactionManager.RegisterInteractionBehaviour(this);
+        }
+      }
+    }
+
+    protected virtual void OnValidate() {
+      _interactionManagerIsNull = interactionManager == null;
+      _body = GetComponent<Rigidbody>();
+    }
+
+    /// <summary> Return the distance the interaction object is from the given world position. </summary>
+    public abstract float GetDistance(Vector3 worldPosition);
+
+    #region Hovering
+
+    /// <summary> Called per-hand when that hand produces a non-zero hover score for this object. </summary>
+    public abstract void HoverBegin(Hand hand);
+
+    /// <summary> Called per-hand when that hand maintains a non-zero hover score for each frame beyond the first. </summary>
+    public abstract void HoverStay(Hand hand);
```

Comment:
For most of these callbacks, I would like to discuss using a HoverStay(List<Hand> hands) for all hands instead of HoverStay(Hand) for each hand one at a time.  I think this would be positive for a number of reasons:

 - If I want to do an action for every frame that an object is hovered, I just put my logic in HoverStay, and it works!
 - If I want to do logic that is based around all currently hovering hands, I also can just use HoverStay, and it's easy
 - I think that it would reduce the number of fields in this class, since a fair number of them are simply temporary state fields used during Stay callbacks to do a calculation like closest hand.

Also, I think it would be useful for the Stay callback of a hand/event to be called on the *same* frame that the Begin callback happened.  If I want to do logic for every frame that an object is hovered, I want to be able to put that logic in HoverStay without getting a 1 frame delay.